lib/Controller/DocumentController.php

```diff
@@ -116,6 +116,13 @@ public function index($fileId) {
 				'instanceId' => $this->settings->getSystemValue('instanceid'),
 			];
 
+			// Update the current file to be accessible with system public shared key
+			$owner = $item->getOwner()->getUID();
+			$absPath = '/' . $owner . '/' .  $item->getInternalPath();
+			$accessList = \OC::$server->getEncryptionFilesHelper()->getAccessList($absPath);
+			$accessList['public'] = true;
+			\OC::$server->getEncryptionManager()->getEncryptionModule()->update($absPath, $owner, $accessList);
```

Comment:
When you have a second encryption module, you should change the API, it is not possible to solve this problem at the app level. We need to keep this in mind. But for now, I think this patch is good to be merged. Thanks for your input.